xmpp/transports.py

```diff
@@ -384,7 +384,10 @@ def _startSSL(self):
         """ Immidiatedly switch socket to TLS mode. Used internally."""
         """ Here we should switch pending_data to hint mode."""
         tcpsock=self._owner.Connection
-        tcpsock._sslObj    = ssl.wrap_socket(tcpsock._sock, None, None)
+        context=ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+        context.check_hostname = False
```

Comment:
It strikes me that xmppy should by default do TLS validation, and perhaps have an option not to, for people dealing with  their own servers that are not set up correctly.   It may be that this code change replicates the old behavior, in which my comment is unrelated to your PR.

Reply:
I tried to make the minimal changes to get the library work in python 3.12, as some ssl library functions were removed and I think this is equivalent to previous code. I'm missing something?